pkg/liqo-controller-manager/namespace-controller/namespace_controller.go

```diff
@@ -0,0 +1,172 @@
+		} else {
+
+			klog.Infof("---------------------- Watch for virtual-nodes labels")
+
+			// 2.b Iterate on all nodes labels (next step only virtual nodes), if the namespace has all the requested
+			// labels, is necessary to create the remote namespace on the remote cluster associated with the virtual
+			// node
+
+			nodes := &v1.NodeList{}
+			if err := r.List(context.Background(), nodes, client.MatchingLabels{"type": "virtual-node"}); err != nil {
+				//if err := r.List(context.Background(), nodes); err != nil {
+				klog.Error(err, "Unable to list virtual nodes")
+				return ctrl.Result{}, err
+			}
+
+			for _, node := range nodes.Items {
+				i := 0
+				found := false
+				dim := len(node.Labels)
+				for key := range node.Labels {
+					i++
+					// I have to check only the offloading.liqo.io/ labels
+					// virtual nodes will always have these offloading.liqo.io/ labels
+					if len(key) > 18 && key[0:19] == offloadingPrefixLabel {
+						if _, ok := labels[key]; !ok {
+							break
+						} else {
+							found = true // found will be no more necessary with only virtual nodes
+						}
+					}
+
+					if i == dim && found {
+						klog.Infof("---------------------- Create namespace for that remote cluter")
+						// TODO : remove from "erase" this virtual node
+					}
+
+				}
+			}
+		}
+
+	}
+
+	if erase {
+		klog.Infof("---------------------- Delete all unnecessary mapping in NNT")
+	}
+
+	klog.Flush()
+	return ctrl.Result{}, nil
+}
+
+// This function is useful only if we decide to accept the renaming of namespaces in "mapping.liqo.io" label.
+// If we don't want to change the name of namespaces, in function reconcile when we check that the name in the NNT is
+// different from the new name, with client we change the value of the new inserted label
+func mappingLabelUpdate(old map[string]string, new map[string]string) bool {
+	ret := false
+	if val1, ok := old[mappingLabel]; ok {
+		ret = val1 != new[mappingLabel]
+	}
+	return ret
+}
+
+// only a simple check in order to avoid useless overhead
+func mappingLabelPresence(labels map[string]string) bool {
+	_, ok := labels[mappingLabel]
+	return ok
+}
+
+// Events not filtered:
+// 1 -- add/delete labels, and mappingLabel is present before or after the update
+// 2 -- update the value of mappingLabel label only
+// 3 -- add namespace with at least mappingLabel
+func manageLabelPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldLabels := e.MetaOld.GetLabels()
+			newLabels := e.MetaNew.GetLabels()
+			return ((len(oldLabels) != len(newLabels)) && (mappingLabelPresence(oldLabels) || mappingLabelPresence(newLabels))) || mappingLabelUpdate(oldLabels, newLabels)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false // i want to use only finalizer
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return mappingLabelPresence(e.Meta.GetLabels())
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
```

Comment:
Instead of setting it explicitly, I wouldn't set any handler for `GenericFunction`

Reply:
so is better to not filter this kind of `GenericEvent` ?

Reply:
yep, I think so